File: pyton_files/ex12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, other):
        if isinstance(other, Produto):
            return self.preco + other.preco
        return NotImplemented

    def __repr__(self):
        return f"Produto({self.nome}, {self.preco})"


produto1 = Produto("Produto A", 60.0)
produto2 = Produto("Produto B", 25.0)
preco_total = produto1 + produto2
print(f"Preço total: {preco_total}")  

File: go_files/ex12.go
package main

import (
	"fmt"
)

type Produto struct {
	nome  string
	preco float64
}

func (p Produto) Somar(outro Produto) Produto {
	return Produto{
		nome:  p.nome + " e " + outro.nome,
		preco: p.preco + outro.preco,
	}
}
func (p Produto) String() string {
	return fmt.Sprintf("Produto: %s, Preço: R$%.2f", p.nome, p.preco)
}

func main() {
	produto1 := Produto{"Produto1", 80.00}
	produto2 := Produto{"Produto2", 10.00}

	produtoSoma := produto1.Somar(produto2)
	fmt.Println(produtoSoma.String())
}
